child_compassion/models/child_pictures.py

```diff
@@ -88,67 +90,73 @@ def create(self, vals):
         if existing_picture:
             return existing_picture
 
-        pictures = super().create(vals)
-        # Retrieve Headshot
-        image_date = pictures._get_picture("Headshot", width=180, height=180)
-        # Retrieve Fullshot
-        image_date = image_date and pictures._get_picture(
-            "Fullshot", width=800, height=1200
-        )
+        picture = super().create(vals)
+        # Fetch the image from the webservice
+        image_date = self._fetch_and_attach_pictures(picture)
+
+        # Handle the case where the picture cannot be fetched
         if not image_date:
-            # We could not retrieve a picture, we cancel the creation
-            pictures.child_id.message_post(
-                body=_(pictures._error_msg), subject=_("Picture update")
-            )
-            pictures.unlink()
-            return False
+            self._handle_picture_fetch_failure(picture)
 
         # Find if same pictures already exist
-        same_pictures = pictures._find_same_picture()
-        if same_pictures:
-            # That case is not likely to happens, it means that the url has
-            #  changed, while the picture stay unchanged.
-            pictures.child_id.message_post(
-                body=_("The picture was the same"), subject=_("Picture update")
-            )
-            pictures.unlink()
-            return False
+        if picture._find_same_picture():
+            self._handle_duplicate_picture(picture)
 
-        pictures.write({"date": image_date})
-        return pictures
+        picture.write({"date": image_date})
+        return picture
 
     ##########################################################################
     #                             PRIVATE METHODS                            #
     ##########################################################################
+    def _fetch_and_attach_pictures(self, picture):
+        """Fetch the pictures from the webservice and attach them to the record."""
+        headshot_fetched = picture._get_picture("Headshot", width=180, height=180)
+        fullshot_fetched = picture._get_picture("Fullshot", width=800, height=1200)
+        return headshot_fetched and fullshot_fetched
+
+    def _handle_picture_fetch_failure(self, picture):
+        """Handle the failure of fetching the picture from the webservice."""
+        message = _("Image cannot be fetched: No image URL available.")
+        logger.warning(message)
+        picture.child_id.message_post(body=message, subject=_("Picture update"))
+        picture.unlink()
+        raise UserError(message)
```

Comment:
We don't want to raise errors, otherwise all pending operations are cancelled.

```suggestion
        self.child_id.message_post(body=message, subject=_("Picture update"))
        self.unlink()
```